src/main/java/pl/javastart/task/Processor.java

```diff
@@ -0,0 +1,41 @@
+package pl.javastart.task;
+
+public class Processor extends Component implements Overclockable {
+    private int temperature;
+    private int clockProcessor;
+    private int clockProcessorAdd;
+    private static final int MAXTEMPERATURE = 300;
+
+    public Processor(String modelName, String manufacturer, String serialNumber) {
+        super(modelName, manufacturer, serialNumber);
+        this.clockProcessor = 3000;
+        this.temperature = 200;
+    }
+
+    public void setClockProcessorAdd(int clockProcessorAdd) {
+        this.clockProcessorAdd = clockProcessorAdd;
+    }
+
+    public void overclock() throws ComponentOverheating {
+        int temperatureIncrease = clockProcessorAdd / 100 * 10;
```

Comment:
Przy zwiększaniu temperatury np. o 90 MHz komponent można podkręcać bez końca i temperatura nie rośnie przez operacje całkowitoliczbowe.
To samo dotyczy pamięci